react-frontend/src/pages/MakeOfferPage.js

```diff
@@ -1,37 +1,101 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { Container, Form, OfferGroup, Offer, HeaderLabel, MakeOfferButton } from '../authentication/AuthenticationStyling';
 import { useNavigate, useParams } from 'react-router-dom';
+import Talk from 'talkjs';
 import axios from 'axios';
 
+const TALKJS_APP_ID = 'tEOH06eA';
+
 function MakeOfferPage() {
-  const [offer, setOffer] = useState("0");  // Initial state for the offer input
+  const [offer, setOffer] = useState("0");
+  const [buyer, setBuyer] = useState(null);
   const navigate = useNavigate();
   const { listingID } = useParams();
 
+  useEffect(() => {
+    const fetchUserProfile = async () => {
+      const token = localStorage.getItem('token');
+      if (!token) {
+        navigate('/login');
+        return;
+      }
+
+      try {
+        const response = await axios.get(`${process.env.REACT_APP_BACKEND_LINK}/users/profile`, {
+          headers: { 'Authorization': `Bearer ${token}` },
+        });
+        setBuyer(response.data);
```

Comment:
this might need to be response.data[0]

Reply:
It might. A good amount of this code needs to get fixed